drivers/adc/adc_stm32.c

```diff
@@ -1527,6 +1605,7 @@ static int adc_stm32_channel_setup(const struct device *dev,
 #if defined(CONFIG_SOC_SERIES_STM32H5X) || ANY_ADC_HAS_DIFFERENTIAL_SUPPORT || \
 	(ANY_ADC_HAS_CHANNEL_PRESELECTION && CONFIG_ADC_STM32_INJECTED_CHANNELS)
 	ADC_TypeDef *adc = config->base;
+	int err;
```

Comment:
I think if only `CONFIG_SOC_SERIES_STM32H5X` is true among above conditions, `err` will trigger an unused variable build warning.